tests/test_repository_lib.py

```diff
@@ -387,6 +369,34 @@ def test_generate_targets_metadata(self):
                                          False)
     self.assertTrue(tuf.formats.TARGETS_SCHEMA.matches(targets_metadata))
 
+    # Test update in targets' delegations
+    keystore_path = os.path.join('repository_data', 'keystore')
+    targets_public_keypath = os.path.join(keystore_path, 'targets_key.pub')
+    targets_public_key = securesystemslib.interface.\
+        import_ed25519_publickey_from_file(targets_public_keypath)
+
+    # Add new key and threshold to delegated role
+    repository.targets('role1').add_verification_key(targets_public_key)
+    repository.targets('role1').threshold = 2
+    role1_keyids = tuf.roledb.get_role_keyids('role1')
+    role1_threshold = tuf.roledb.get_role_threshold('role1')
+    roleinfo = tuf.roledb.get_roleinfo('targets')
+    delegations = roleinfo['delegations']
+    old_delegations = copy.deepcopy(delegations)
+
+    targets_metadata  = \
+      repo_lib.generate_targets_metadata(targets_directory, target_files,
+                                         version, expiration_date, delegations,
+                                         False)
```

Comment:
I realise you've just copied other examples in this file, but it's more inline with the coding style to do something like:
```suggestion
    targets_metadata  = repo_lib.generate_targets_metadata(targets_directory,
        target_files, version, expiration_date, delegations, False)
```